app/controllers/jobseekers/job_applications_controller.rb

```diff
@@ -25,6 +25,14 @@ def review
     session[:back_to_review] = (session[:back_to_review] || []).push(job_application.id).uniq
   end
 
+  # rubocop:disable Style/GuardClause
+  def about_your_application
+    if profile.nil? || profile&.personal_details&.right_to_work_in_uk?
+      redirect_to new_quick_apply_jobseekers_job_job_application_path(vacancy.id)
+    end
+  end
+  # rubocop:enable Style/GuardClause
+
```

Comment:
Question :raising_hand_man: wouldn't this cause a double redirection when the `about_your_application` step is not needed? 
`new -> about_your_application -> new_quick_apply...`

In that case, have you considered grouping all the redirection logic together in the `#new` method?
It would either redirect to `A` or `B` instead of redirecting to `A` and then redirecting to `B` if `A` wasn't necessary.
```ruby
  def new
    request_event.trigger(:vacancy_apply_clicked, vacancy_id: vacancy.id)
    send_dfe_analytics_event

    return unless quick_apply?

    if profile.nil? || profile&.personal_details&.right_to_work_in_uk?
      redirect_to new_quick_apply_jobseekers_job_job_application_path(vacancy.id)
    else
      redirect_to about_your_application_jobseekers_job_job_application_path(vacancy.id)
    end
  end
```

Reply:
Fair enough!